django/app_tutorial/models.py

```diff
@@ -4,43 +4,80 @@
 from django.contrib.auth.models import User
 
 
-class Nationality(models.Model):
-    abbreviation = models.CharField(max_length=5)
-    name = models.CharField(max_length=75)
+class DivisionAge(models.Model):
+    name = models.CharField(max_length=50)
+    # In our discussion this was denoted as type, I thought this might be handier.
+    age_min = models.IntegerField()
+    age_max = models.IntegerField()
 
 
-class Division(models.Model):
+class DivisionType(models.Model):
+    # TODO (Feedback): In our discussion this was division gender, but I feel like Mixed and Open aren't really genders.
     name = models.CharField(max_length=50)
 
 
-class TournamentType(models.Model):
-    name = models.CharField(max_length=50)
+class FieldType(models.Model):
+    type = models.CharField(max_length=50)
 
 
-class GameType(models.Model):
-    name = models.CharField(max_length=50)
+class Field(models.Model):
+    location = models.CharField(max_length=100)
+    type = models.ForeignKey(FieldType)
 
 
-class GameReportEventType(models.Model):
+class Location(models.Model):
     name = models.CharField(max_length=50)
+    location = models.CharField(max_length=100)
 
 
-class SpiritScoreCategory(models.Model):
-    name = models.CharField(max_length=50)
+class Tournament(models.Model):
+    name = models.CharField(max_length=75)
+    location = models.ForeignKey(Location)
+    date_begin = models.DateField('Begin Date')
+    date_end = models.DateField('End Date')
+    # type = models.ForeignKey(TournamentType)
 
 
-class Team(models.Model):
+class TournamentDivision(models.Model):
+    tournament = models.ForeignKey(Tournament)
+    age_group = models.ForeignKey(DivisionAge)
+    division = models.ForeignKey(DivisionType)
+
+
+class TournamentField(models.Model):
+    tournament = models.ForeignKey(Tournament)
+    field = models.ForeignKey(Field)
+    field_number = models.IntegerField()  # TODO (Feedback): imo this is way more practical than a string name
+
+
+class Squad(models.Model):
+    # TODO: When teams are added, make name nullable and build squad name from division age and type if name is empty.
```

Comment:
You mean, when a Squad is added (and not team)? 

Overloading the **save()** method seems the way to go but I haven't tried it yet ([overriding-predefined-model-methods](https://docs.djangoproject.com/en/dev/topics/db/models/#overriding-predefined-model-methods)).